Sources/SwiftSyntaxBuilder/ConvenienceInitializers/CustomAttributeConvenienceInitializers.swift

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension CustomAttribute {
+  /// A convenience initializer that allows passing in arguments using a result builder
+  /// and automatically adds parentheses as needed, similar to the convenience
+  /// initializer for `FunctionCallExpr`.
+  public init(
+    _ attributeName: ExpressibleAsTypeBuildable,
+    @TupleExprElementListBuilder argumentList: () -> ExpressibleAsTupleExprElementList? = { nil }
+  ) {
+    let argumentList = argumentList()
+    self.init(
+      attributeName: attributeName,
+      leftParen: argumentList.map { _ in .leftParen },
```

Comment:
Hmm, not sure I like the `.map` here. Personally I would make `argumentList != nil ? .leftParent : nil` etc

Reply:
I like omitting explicit `nil`s where possible, perhaps that is the Haskell programmer in me 😄 If mapping the optional is too obscure, I'd be fine with using a ternary too though.